src/components/layer-controller/reducer.js

```diff
@@ -66,9 +83,38 @@ export default function reducer(channels, action) {
       const layerProps = {
         [propertyToUpdate]: updatedValues,
       };
+      // Publish deck.gl layer props.
       PubSub.publish(LAYER_CHANGE, { layerId, layerProps });
       return nextChannels;
     }
+    case 'CHANGE_GLOBAL_CHANNELS_SELECTION': {
+      const { selection, publish } = payload;
+      // Update channel selection for new state.
+      const nextChannels = Object.assign({},
+        ...Object.keys(channels).map(channelId => (
+          {
+            [channelId]: {
+              ...channels[channelId],
+              selection: {
+                ...channels[channelId].selection,
+                ...selection,
+              },
+            },
+          }
+        )));
```

Comment:
Can you use the same spreading as above (instead of `Object.assign`) for consistency:

```suggestion
     const nextChannels = {
         ...channels,
         [channelId]: {
           ...channels[channelId],
           selection: {
             ...channels[channelId].selection,
             ...selection,
            }
         }
      }
```